### PR TITLE
Airbyte-ci: Use async click

### PR DESCRIPTION
**Pull request by erohmensing**
## What
* Use async click in airbyte ci. Remove usages of `anyio.run()`

## 🚨 User Impact 🚨
None

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- Connector version is set to `0.0.1`
    - `Dockerfile` has version `0.0.1`
- Documentation updated
    - Connector's `README.md`
    - Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - `docs/integrations/<source or destination>/<name>.md` including changelog with an entry for the initial version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - `docs/integrations/README.md`

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Connector Generator</strong></summary>

- Issue acceptance criteria met
- PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/resources/pull-requests-handbook)
- If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:generateScaffolds` then checking in your changes
- Documentation which references the generator is updated as needed

</details>

<details><summary><strong>Updating the Python CDK</strong></summary>

### Airbyter

Before merging:
- Pull Request description explains what problem it is solving
- Code change is unit tested
- Build and my-py check pass
- Smoke test the change on at least one affected connector
   - On Github: Run [this workflow](https://github.com/airbytehq/airbyte/actions/workflows/connectors_tests.yml), passing `--use-local-cdk --name=source-<connector>` as options
   - Locally: `airbyte-ci connectors --use-local-cdk --name=source-<connector> test`
- PR is reviewed and approved
      
After merging:
- [Publish the CDK](https://github.com/airbytehq/airbyte/actions/workflows/publish-cdk-command-manually.yml)
   - The CDK does not follow proper semantic versioning. Choose minor if this the change has significant user impact or is a breaking change. Choose patch otherwise.
   - Write a thoughtful changelog message so we know what was updated.
- Merge the platform PR that was auto-created for updating the Connector Builder's CDK version
   - This step is optional if the change does not affect the connector builder or declarative connectors.

</details>

### CODE REVIEW

File: airbyte-ci/connectors/pipelines/README.md
Comment (bnchrch) [inline airbyte-ci/connectors/pipelines/README.md:401]: ❓

File: airbyte-ci/connectors/pipelines/pipelines/cli/airbyte_ci.py
Comment (bnchrch) [inline airbyte-ci/connectors/pipelines/pipelines/cli/airbyte_ci.py:115]: ❓ why did we have to make this async?

File: airbyte-ci/connectors/pipelines/pyproject.toml
Comment (bnchrch) [inline airbyte-ci/connectors/pipelines/pyproject.toml:27]: We need to bump the version of the package in this file

**Review by bnchrch — CHANGES_REQUESTED** [review]
No issues here!

Just need to bump the version and run a test + prepublish from this branch

File: airbyte-ci/connectors/pipelines/pipelines/cli/airbyte_ci.py
Comment (erohmensing) [reply airbyte-ci/connectors/pipelines/pipelines/cli/airbyte_ci.py:115]: We didn't _have_ to, but we were able to! (In order to remove the `anyio.run`)

Comment (erohmensing) [reply airbyte-ci/connectors/pipelines/pipelines/cli/airbyte_ci.py:115]: E.g. [here](https://github.com/airbytehq/airbyte/pull/31748/files#diff-f35c71638f2ffaabb83f6f5b5aa4c5c8ef5e2baab9968e7aa14fc32c9a7f54d2R112)

Comment (bnchrch) [reply airbyte-ci/connectors/pipelines/pipelines/cli/airbyte_ci.py:115]: Good call!

**Review by bnchrch — APPROVED** [review]
Approving now! But if you could post a link to a successfull GHA run of prepublish and test based on this branch before merging that would be great!

File: airbyte-ci/connectors/pipelines/README.md
Comment (erohmensing) [reply airbyte-ci/connectors/pipelines/README.md]: I was waiting for approval since there were so many open PRs I knew I'd have to change this anyway 😄 it was a reminder to bump the version in pyproject.toml

**Comment by erohmensing** [comment]
Publish run: https://github.com/airbytehq/airbyte/actions/runs/6628185444 ✅ 
Test runs: https://github.com/airbytehq/airbyte/pull/31778 🟡 (pending)

File: airbyte-ci/connectors/pipelines/pyproject.toml
Comment (erohmensing) [inline airbyte-ci/connectors/pipelines/pyproject.toml:31]: This bump was required in order to be able to use `pytest.asyncio`. Confirmed that all tests still run and pass after this

File: airbyte-ci/connectors/pipelines/README.md
Comment (erohmensing) [inline airbyte-ci/connectors/pipelines/README.md:403]:
```suggestion
| 2.3.1   | [#31748](https://github.com/airbytehq/airbyte/pull/31748)  | Use AsyncClick library instead of base Click.                                                             |
```

File: airbyte-ci/connectors/pipelines/pyproject.toml
Comment (erohmensing) [inline airbyte-ci/connectors/pipelines/pyproject.toml:7]:
```suggestion
version = "2.3.1"
```

**Pull request merged**
